### PR TITLE
Show user statistics in an API call

### LINKED ISSUE #2176: API for statistics
**Is your feature request related to a problem? Please describe.**
It'd be great to be able to access different statistics through the API to present them for example in a mobile client software. What I'm currently most interested in are user's (and total) hours this and last month and week and total billing this and last month and week, but obviously it'd be better configurable similarly to the widgets.

**Describe the solution you'd like**
Versatile API call(s) to fetch the needed statistics information by querying an endpoint with the relevant arguments.

### PR DESCRIPTION
**Pull request by ZeiP**
## Description
Adds an API call for user statistics. It's probably not acceptable as-is, but this worked for me so it's at least a start.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] I verified that my code applies to the guidelines (`composer code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [X] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

Fixes #2176.

### CODE REVIEW

File: src/API/ReportController.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Returns stats.
+     *
+     * @SWG\Response(
+     *      response=200,
+     *      description="Returns user stats",
+     *      @SWG\Schema(
+     *          type="array",
+     *          @SWG\Items(ref="#/definitions/ActivityCollection")
```
Comment (kevinpapst) [inline src/API/ReportController.php:59]: wrong result type

File: src/API/ReportController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     *      )
+     * )
+     *
+     * @ApiSecurity(name="apiUser")
+     * @ApiSecurity(name="apiToken")
+     */
+    public function userAction(ParamFetcherInterface $paramFetcher): Response
+    {
+        $currentUser = $this->getUser();
```
Comment (kevinpapst) [inline src/API/ReportController.php:68]: should be limited  by `$this->isGranted('view', $currentUser)`

File: src/API/ReportController.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    private $viewHandler;
+
+    public function __construct(ViewHandlerInterface $viewHandler, TimesheetRepository $timesheetRepository)
+    {
+        $this->timesheetRepository = $timesheetRepository;
+        $this->viewHandler = $viewHandler;
+    }
+
+    /**
+     * Returns stats.
```
Comment (kevinpapst) [inline src/API/ReportController.php:52]: Not the shortest documentation I have ever seen ... but almost 😁

**Review by kevinpapst — CHANGES_REQUESTED** [review]
Thanks!
Tests are missing and I think it might be good to allow to pass in the user ID.
What do you say?

**Comment by kevinpapst** [comment]
Thank you for your PR and the invested time 👍 

Nevertheless, I am closing this PR unmerged for two reasons: 

- I don't want to include API endpoints for statistics
- Review feedback and failing tests ignored

This feature should be published as plugin.

**Comment by ZeiP** [comment]
Sorry, unfortunately I haven't had time to look into this. I'd be happy to fix the PR, but if it's not going to be merged then I'll just keep it in my local branch. I doubt publishing a separate plugin for a single API endpoint is useful.

**Comment by kevinpapst** [comment]
I was working on a reporting plugin. If I manage to publish it (time is missing) I will send you a ping, then we can move this endpoint there.

**Pull request closed without merging**